api/indexer.oas2.json

```diff
@@ -484,6 +484,98 @@
         }
       }
     },
+    "/v2/applications/{application-id}/boxes": {
+      "get": {
+        "description": "Given an application ID, returns the box names of that application sorted lexicographically.",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "search"
+        ],
+        "operationId": "searchForApplicationBoxes",
+        "summary": "Get box names for a given application.",
+        "parameters": [
+          {
+            "type": "integer",
+            "name": "application-id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "$ref": "#/parameters/limit"
+          },
+          {
+            "type": "string",
+            "description": "The next page of results starting from a box name in goal-arg form 'encoding:value'. Use the next token provided by the previous results.",
+            "name": "next",
+            "in": "query",
+            "x-algorand-format": "GOALarg"
```

Comment:
I'm creating this out of "thin air". Should I document in a more transparent fashion somewhere?

Reply:
@jannotti  ^^^ ?

Reply:
Is the question where it should be documented? (For that I know nothing with respect to indexer) or for feedback on the interface? My read on that use that we shouldn't need to introduce a token and next mechanism because boxes are inherently ordered. The API can be "give one page of results with all boxes after name `name`". Clients can cycle through them all by sending "" first, and then the last box name from their previous query until an unfilled page is returned. This removes the need to maintain any state at the server.

But then again, if this exact technique has already been used, we could stick with it.

Reply:
My question is confusing. The paging happening  here is stateless as you suggest and conforms with other indexder endpoints. What is "out of thin air" here is this `"x-algorand-format": "GOALarg"` which I'm proposing to indicate the encoding scheme.

I don't know whether this should be documented in some way, or this self-documentation is good 'nuf.

@winder - any thoughts as well?

Reply:
Oh, yes, I see. How did we document that in `algod`? We have a similar scheme for the box lookup.

Reply:
I copied the wording from [algod's swagger](https://github.dev/algorand/go-algorand/blob/feature/avm-box/daemon/algod/api/algod.oas2.json#L1377) but I don't know how else we're communicating this. I don't think this is urgent, we can wait till Will comes back from vacay.

Reply:
That was the wrong link. Here's the updated one: https://github.com/algorand/go-algorand/blob/4688f93ac9bbbb346a871f08602b1fce20ee3d58/daemon/algod/api/algod.oas2.json#L1501

Reply:
Seems like we should use the same language, so this seems right for now.  Worth noting that if someone complains and changes, both should be changed.

Reply:
@tzaffi Given my current understanding, I prefer to remove `"x-algorand-format": "GOALarg"`.

`x-algorand-format` carries special semantics in _generator_: 
 https://github.com/algorand/generator/search?q=x-algorand-format.  By adding `GOALarg`, it falsely implies _generator_ produces models differently.